File: codeForces/contests/1000div2/D.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n, m;
        cin >> n >> m;

        vector<int> a(n), b(m);
        for (int i = 0; i < n; ++i) cin >> a[i];
        for (int i = 0; i < m; ++i) cin >> b[i];

        int kmax;
        if (n > 2 * m) {
            kmax = m;
        } else if (m > 2 * n) {
            kmax = n;
        } else {
            kmax = (n + m) / 3;
        }
        cout << kmax << "\n";

        sort(a.begin(), a.end());
        sort(b.begin(), b.end());

        vector<int> adre(a.begin() + n / 2, a.end());
        vector<int> aesq(a.begin(), a.begin() + n / 2);
        reverse(aesq.begin(), aesq.end());

        vector<int> bdre(b.begin() + m / 2, b.end());
        vector<int> besq(b.begin(), b.begin() + m / 2);
        reverse(besq.begin(), besq.end());

        for (int i = 1; i <= kmax; ++i) {
            vector<int> adre2 = adre, aesq2 = aesq, bdre2 = bdre, besq2 = besq;
            int n2 = n, m2 = m, ans = 0;

            for (int e = 0; e < i; ++e) {
                if (n2 == i - e) {
                    ans += bdre2.back() - besq2.back();
                    bdre2.pop_back();
                    besq2.pop_back();
                    n2 -= 1;
                    m2 -= 2;
                } else if (m2 == i - e) {
                    ans += adre2.back() - aesq2.back();
                    adre2.pop_back();
                    aesq2.pop_back();
                    n2 -= 2;
                    m2 -= 1;
                } else if (bdre2.back() - besq2.back() > adre2.back() - aesq2.back()) {
                    ans += bdre2.back() - besq2.back();
                    bdre2.pop_back();
                    besq2.pop_back();
                    n2 -= 1;
                    m2 -= 2;
                } else {
                    ans += adre2.back() - aesq2.back();
                    adre2.pop_back();
                    aesq2.pop_back();
                    n2 -= 2;
                    m2 -= 1;
                }
            }
            cout << ans << " ";
        }
        cout << "\n";
    }

    return 0;
}

File: codeForces/contests/1000div2/D.py

for _ in range(int(input())):

    n, m = list(map(int, input().split()))
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))

    if n > 2*m:
        kmax = m
    elif m > 2*n:
        kmax = n
    else:
        kmax = (n+m)//3
    print(kmax)

    a.sort()
    b.sort()
    adre = a[n//2:]
    aesq = a[:n//2]
    aesq = aesq[::-1]
    bdre = b[m//2:]
    besq = b[:m//2]
    besq = besq[::-1]

    for i in range(1, kmax+1):
        adre2, aesq2, bdre2, besq2 = adre[:], aesq[:], bdre[:], besq[:]
        n2, m2 = n, m
        ans = 0
        for e in range(i):
            if n2 == i-e:
                ans += bdre2.pop() - besq2.pop()
                n2 -= 1
                m2 -= 2
            elif m2 == i-e:
                ans += adre2.pop() - aesq2.pop()
                n2 -= 2
                m2 -= 1
            elif bdre2[-1] - besq2[-1] > adre2[-1] - aesq2[-1]:
                ans += bdre2.pop() - besq2.pop()
                n2 -= 1
                m2 -= 2
            else:
                ans += adre2.pop() - aesq2.pop()
                n2 -= 2
                m2 -= 1
        print(ans, end = " ")
    print()
